source/game_sa/Entity/Vehicle/CBoat.cpp

```diff
@@ -1,31 +1,141 @@
     ReversibleHooks::Install("CBoat", "ProcessOpenDoor", 0x6F0190, &CBoat::ProcessOpenDoor_Reversed);
     ReversibleHooks::Install("CBoat", "BlowUpCar", 0x6F21B0, &CBoat::BlowUpCar_Reversed);
 
     //Class methods
     ReversibleHooks::Install("CBoat", "PruneWakeTrail", 0x6F0E20, &CBoat::PruneWakeTrail);
     ReversibleHooks::Install("CBoat", "AddWakePoint", 0x6F2550, &CBoat::AddWakePoint);
+    ReversibleHooks::Install("CBoat", "SetupModelNodes", 0x6F01A0, &CBoat::SetupModelNodes);
+    ReversibleHooks::Install("CBoat", "DebugCode", 0x6F0D00, &CBoat::DebugCode);
+    ReversibleHooks::Install("CBoat", "ModifyHandlingValue", 0x6F0DE0, &CBoat::ModifyHandlingValue);
+    ReversibleHooks::Install("CBoat", "PrintThrustAndRudderInfo", 0x6F0D90, &CBoat::PrintThrustAndRudderInfo);
 
     //Other
+    ReversibleHooks::Install("CBoat", "FillBoatList", 0x6F2710, &CBoat::FillBoatList);
+    ReversibleHooks::Install("CBoat", "IsSectorAffectedByWake", 0x6F0E80, &CBoat::IsSectorAffectedByWake);
+    ReversibleHooks::Install("CBoat", "IsVertexAffectedByWake", 0x6F0F50, &CBoat::IsVertexAffectedByWake);
+    ReversibleHooks::Install("CBoat", "CheckForSkippingCalculations", 0x6F10C0, &CBoat::CheckForSkippingCalculations);
     ReversibleHooks::Install("CBoat", "GetBoatAtomicObjectCB", 0x6F00D0, &GetBoatAtomicObjectCB);
 }
 
+CBoat::CBoat(int modelIndex, unsigned char createdBy) : CVehicle(createdBy)
+{
+    memset(&m_boatFlap, 0, sizeof(m_boatFlap));
+    auto pModelInfo = reinterpret_cast<CVehicleModelInfo*>(CModelInfo::GetModelInfo(modelIndex));
+    auto iHandlingId = pModelInfo->m_nHandlingId;
+    m_nVehicleSubClass = eVehicleType::VEHICLE_BOAT;
+    m_nVehicleClass = eVehicleType::VEHICLE_BOAT;
+    m_vecBoatMoveForce.Set(0.0F, 0.0F, 0.0F);
+    m_vecBoatTurnForce.Set(0.0F, 0.0F, 0.0F);
+    m_nPadNumber = 0;
+    m_fMovingHiRotation = 0.0;
+    m_fPropSpeed = 0.0;
+    m_fPropRotation = 0.0;
+    m_nAttackPlayerTime = CTimer::m_snTimeInMilliseconds;
+    CVehicle::SetModelIndex(modelIndex);
+    CBoat::SetupModelNodes();
+
+    m_pHandlingData = &gHandlingDataMgr.m_aVehicleHandling[iHandlingId];
+    m_nHandlingFlagsIntValue = m_pHandlingData->m_nHandlingFlags;
+    m_pFlyingHandlingData = gHandlingDataMgr.GetFlyingPointer(iHandlingId);
+    m_pBoatHandling = gHandlingDataMgr.GetBoatPointer(iHandlingId);
+
+    pModelInfo->ChooseVehicleColour(m_nPrimaryColor, m_nSecondaryColor, m_nTertiaryColor, m_nQuaternaryColor, 1);
+
+    m_fMass = m_pHandlingData->m_fMass;
+    m_fTurnMass = m_pHandlingData->m_fTurnMass * 0.5;
+    m_vecCentreOfMass = m_pHandlingData->m_vecCentreOfMass;
+    m_fElasticity = 0.1;
+    m_fBuoyancyConstant = m_pHandlingData->m_fBuoyancyConstant;
+
+    if (m_pHandlingData->m_fDragMult <= 0.01F)
+        m_fAirResistance = m_pHandlingData->m_fDragMult;
+    else
+        m_fAirResistance = m_pHandlingData->m_fDragMult / 1000.0 * 0.5;
```

Comment:
```suggestion
    else
        m_fAirResistance = m_pHandlingData->m_fDragMult / 1000.0f * 0.5f;
```
There are way too many warnings generated from your PR because you are mixing floats with doubles. There are way too many warnings, so please try to get rid of them.
![image](https://user-images.githubusercontent.com/10183157/103786731-93197880-505e-11eb-8e2b-3f2edf118d0b.png)

Reply:
Fixed most warnings in places that i could